brave/brave-core
# Add setting to restrict the resolution of .onion to Tor Windows

Resolves brave/brave-browser#33735 and https://github.com/brave/brave-browser/issues/34076

The original feature was added in https://github.com/brave/brave-core/pull/19621



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Start the browser with a fresh profile.
2. Open a **normal window**.
3. Go to `brave://settings/privacy` and confirm thatenable _Only resolve .onion in Tor windows_.
4. Open the devtols in the console tab.
5. Open <https://fmarier.github.io/brave-testing/onion-subresource-33735.html> to see a broken image.
7. Confirm that the error for `onion16.png` is `net::ERR_BLOCKED_BY_CLIENT`.
8. Go to `brave://settings/privacy` and disable _Only resolve .onion in Tor windows_.
9. Clear the devtools console.
10. Force-refresh the page to see the same broken image.
11. Confirm that the error for `onion16.png` is `net::ERR_NAME_NOT_RESOLVED`.
12. Repeat these steps in a **private window with Tor**.
13. Confirm that the image is loading correctly in a Tor window regardless of the value of _Only resolve .onion in Tor windows_.

Finally, check that the first setting under _Tor windows_ in `brave://settings/privacy` has a _Learn more_ link which points to <https://support.brave.com/hc/en-us/articles/360018121491-What-is-a-Private-Window-with-Tor-Connectivity->.
